force-app/main/default/classes/SOQL.cls

```diff
@@ -28,9 +28,11 @@ public with sharing class SOQL {
      * @return List of all leads with Id, LastName, Company, Phone, Email
      * Update the corresponding test class and testGetAllLeads method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+    public static List<Lead> getAllLeads (List<String> leadsByLastName) {
+        List<Lead> allLeads = [SELECT Id, LastName, Company, Phone, Email FROM Lead WHERE LastName IN :leadsByLastName];
+        return allLeads;
```

Comment:
Something to try out:
```suggestion
        return [SELECT Id, LastName, Company, Phone, Email FROM Lead WHERE LastName IN :lastNames];
```

Reply:
Thanks Chase. This makes sense. As I'm not using the List anywhere, there is no point of returning a List here, instead I can directly return the records avoiding the List creation.